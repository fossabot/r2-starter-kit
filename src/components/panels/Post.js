import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import Loading from 'components/elements/Loading';
import Breadcrumb from 'containers/blocks/Breadcrumb';
import Content from 'containers/blocks/Content';

class Post extends PureComponent {
  static propTypes = {
    uiState: PropTypes.oneOf(['loading', 'idle', 'failure']),
  };
  static defaultProps = {
    uiState: 'idle',
  }
  render() {
    const { uiState } = this.props;
    return (
      <div>
        {uiState === 'loading' && <Loading />}
        {uiState === 'idle' &&
          <div className="ts text container">
            <Breadcrumb />
            <div className="ts divider" />
            <Content />
          </div>
        }
      </div>
    );
  }
}

export default Post;
