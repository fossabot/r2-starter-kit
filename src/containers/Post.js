import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import { getPost } from 'redux/modules/post';
import Post from 'components/panels/Post';

function mapStateToProps(state, ownProps) {
  return {
    id: parseInt(ownProps.match.params.id, 10),
    post: state.post,
    state: state.state,
  };
}
function mapDispatchToProps(dispatch) {
  return bindActionCreators({ getPost }, dispatch);
}

export default connect(mapStateToProps, mapDispatchToProps)(Post);
