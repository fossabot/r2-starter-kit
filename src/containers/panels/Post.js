import { compose } from 'redux';
import { connect } from 'react-redux';

import Post from 'components/panels/Post';
import { withLifecycle } from 'components/hoc';
import { handleDidMount } from 'sagas/panels/post';

function mapStateToProps(state) {
  return state;
}

function mapDispatchToProps(dispatch) {
  return { dispatch };
}

function mergeProps(state, { dispatch }, ownProps) {
  return {
    ...ownProps,

    // withLifecycle
    handleDidMount: () => dispatch(handleDidMount()),

    // custom props
    uiState: state.ui.state.data,
  };
}

export default compose(
  connect(mapStateToProps, mapDispatchToProps, mergeProps),
  withLifecycle(),
)(Post);
