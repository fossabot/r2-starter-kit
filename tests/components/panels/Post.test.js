import React from 'react';
import { shallow } from 'enzyme';

import Post from 'components/panels/Post';

describe('<Post />', () => {
  const initProps = {
    id: null,
    post: {},
    state: 'loading',

    getPost: jest.fn(),
  };

  let wrapper;
  beforeAll(() => {
    wrapper = shallow(<Post {...initProps} />);
  });

  it('should call getPost in componentDidMount', () => {
    expect(initProps.getPost).toHaveBeenCalledTimes(1);
  });

  it('should render loading page', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render post content page', () => {
    wrapper.setProps({
      id: 1,
      post: {
        title: 'Post Title',
        author: {
          name: 'Tony Pai',
          email: 'tonypai@me.com',
        },
        body: 'Post Content',
      },
      state: 'idle',
    });
    expect(wrapper).toMatchSnapshot();
  });
});
