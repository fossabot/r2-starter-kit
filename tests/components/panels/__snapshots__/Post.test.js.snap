// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Post /> should render loading page 1`] = `
<div>
  <Loading />
  <div
    className="ts text container"
  >
    <Breadcrumb
      data={
        Array [
          Object {
            "link": "/",
            "text": "Post List",
          },
          Object {
            "text": undefined,
          },
        ]
      }
    />
    <div
      className="ts divider"
    />
  </div>
</div>
`;

exports[`<Post /> should render post content page 1`] = `
<div>
  <div
    className="ts text container"
  >
    <Breadcrumb
      data={
        Array [
          Object {
            "link": "/",
            "text": "Post List",
          },
          Object {
            "text": "Post Title",
          },
        ]
      }
    />
    <div
      className="ts divider"
    />
    <Content
      author={
        Object {
          "email": "tonypai@me.com",
          "name": "Tony Pai",
        }
      }
      body="Post Content"
      id={1}
      title="Post Title"
    />
  </div>
</div>
`;
